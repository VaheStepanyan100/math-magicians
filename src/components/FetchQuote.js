const { useEffect, useState } = require('react');

const FetchQuote = () => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      try {
        setLoading(true);
        const response = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=imagination',
          {
            headers: {
              'X-Api-Key': 'vZdI6eGz2cS+jTb71dvouA==X8Xd3lMbKviCRmqW',
            },
          },
        );
        if (!response.ok) {
          throw new Error('Something went wrong!');
        }
        const data = await response.json();
        setData(data);
        setLoading(false);
      } catch (error) {
        setError(true);
        setLoading(false);
      }
    };

    fetchData();
  }, []);

  if (error) {
    return <p>Something went wrong!</p>;
  }
  if (loading) return <p>Loading...</p>;

  return (
    <div className="quotes">
      {data.map((item) => (
        <p key={1}>
          <p>{item.quote}</p>
          {' '}
          <p>
            -
            {' '}
            {item.author}
          </p>
        </p>
      ))}
    </div>
  );
};

export default FetchQuote;
