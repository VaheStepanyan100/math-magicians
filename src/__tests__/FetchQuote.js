import { render } from '@testing-library/react';
import FetchQuote from '../components/FetchQuote';

test('renders FetchQuote correctly', () => {
  const { container } = render(<FetchQuote />);
  expect(container).toMatchSnapshot();
});
