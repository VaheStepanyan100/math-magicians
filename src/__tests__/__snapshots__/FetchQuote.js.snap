// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FetchQuote correctly 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
